api/src/main/java/com/messagebird/MessageBirdClient.java

```diff
@@ -1767,4 +1771,54 @@ public void cancelNumber(String number) throws UnauthorizedException, GeneralExc
         final String url = String.format("%s/phone-numbers", NUMBERS_CALLS_BASE_URL);
         messageBirdService.deleteByID(url, number);
     }
+
+    /**
+     * Uploads a file and returns the assigned ID.
+     *
+     * @param binary the bytes of the file to upload.
+     * @param contentType the content type of the file (e.g. "image/png").
+     * @return FileUploadResponse
+     * @throws GeneralException      general exception
+     * @throws UnauthorizedException if client is unauthorized
+     * @see #downloadFile
+     */
+    public FileUploadResponse uploadFile(byte[] binary, String contentType) throws GeneralException, UnauthorizedException {
+        if (binary == null) {
+            throw new IllegalArgumentException("File binary must be specified.");
+        }
+        if (contentType == null) {
+            throw new IllegalArgumentException("Content type must be specified.");
+        }
+
+        final String url = MESSAGING_BASE_URL + FILES_PATH;
+        final Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", contentType);
+        return messageBirdService.sendPayLoad("POST", url, headers, binary, FileUploadResponse.class);
+    }
+
+    /**
+     * Downloads a file and stores it with the provided filename in the basePath directory. The
+     * basePath may be null. If basePath is null, the default download directory will be the
+     * /Download folder in the user home directory.
+     *
+     * @param id       the ID of the file, provided when the file was uploaded
+     * @param basePath store location. It should be a directory. Property is nullable if $HOME is accessible
+     * @param filename the name of the file to download to.
+     * @return the path where the downloaded file is stored
+     * @throws NotFoundException     if the file does not exist
+     * @throws GeneralException      general exception
+     * @throws UnauthorizedException if client is unauthorized
+     * @see #uploadFile
+     */
+    public String downloadFile(String id, String filename, String basePath) throws GeneralException, UnauthorizedException, NotFoundException {
+        if (id == null) {
+            throw new IllegalArgumentException("File ID must be specified.");
+        }
+        if (filename == null) {
```

Comment:
filename param is not a mandatory field. Let's keep the ability to set but remove the exception. If it is null, use file id as a file name.